modus/agents.mdx

```diff
@@ -0,0 +1,634 @@
+    )
+
+    output, _ := model.Invoke(input)
+    analysis := output.Choices[0].Message.Content
+
+    // Update confidence based on data volume
+    i.confidence = float64(len(i.briefings)) / 10.0
+    if i.confidence > 1.0 {
+        i.confidence = 1.0
+    }
+
+    result := fmt.Sprintf(`Analysis complete:
+        %s
+
+        (Confidence: %.2f based on %d briefings)`,
+        analysis,
+        i.confidence,
+        len(i.briefings))
+    return &result, nil
+}
+
+func (i *IntelligenceAgent) getThreatAssessment() (*string, error) {
+    if len(i.briefings) == 0 {
+        result := `No intelligence processed yet.
+            Deploy to field for analysis.`
+        return &result, nil
+    }
+
+    result := fmt.Sprintf(`Current threat assessment:
+        %d briefings analyzed.
+
+        Confidence level: %.2f.
+        Agent operational.`,
+        len(i.briefings),
+        i.confidence)
+    return &result, nil
+}
+```
+
+What makes this operationally superior:
+
+- No transmission of massive state files on every contact
+- No reconstruction of intelligence history from scratch
+- AI models see complete operational context, not just current input
+- No risk of losing analytical data during communication failures
+- Intelligence quality improves with each briefing processed
+
+The agent maintains its complete operational state and intelligence database in
+secure memory, accessing it instantly when processing new directives and
+invoking AI models.
+
+### Long-running mission operations
+
+For complex multi-phase operations, agents can update their state continuously
+while providing real-time progress tracking:
+
+```go
+func (i *IntelligenceAgent) analyzeBackgroundIntel(data *string) (*string, error) {
+    // Initialize multi-phase operation
+    i.currentMission = &MissionPhase{
+        Name:      "Deep Intelligence Analysis",
+        StartTime: time.Now(),
+        Complete:  false,
+    }
+    i.missionLog = []string{"Mission initiated: Deep intelligence analysis"}
+
+    // Phase 1: Data Processing
+    i.updateMissionLog("Phase 1: Processing raw intelligence data...")
+    i.briefings = append(i.briefings, *data)
+
+    // Phase 2: AI Analysis (simplified for brevity)
+    i.updateMissionLog("Phase 2: Generating threat assessment...")
+    // ... AI model invocation ...
+
+    // Phase 3: Mission Complete
+    i.currentMission.Complete = true
+    i.currentMission.Duration = time.Since(i.currentMission.StartTime)
+    i.updateMissionLog("Mission completed successfully")
+
+    return &result, nil
+}
+
+func (i *IntelligenceAgent) updateMissionLog(message string) {
+    timestamp := time.Now().Format("15:04:05")
+    logEntry := fmt.Sprintf("[%s] %s", timestamp, message)
+    i.missionLog = append(i.missionLog, logEntry)
+}
+```
+
+This demonstrates how agents maintain state across complex, multi-step
+operations while providing detailed progress tracking for monitoring systems.
+
+## Monitoring ongoing operations
```

Comment:
@ryanfoxtyler @mattjohnsonpint - keen for more examples in Stream in and Stream out, and what the GraphQL Subscriptions protocol and API looks like - ideally we reaplce polling, in this example, with a subscription.